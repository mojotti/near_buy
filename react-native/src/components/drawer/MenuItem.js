import React, { Component } from 'react';
import { Text, TouchableOpacity, View } from 'react-native';
import { NavigationActions } from 'react-navigation';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Icon from 'react-native-vector-icons/MaterialIcons';
import { styles } from '../../static/styles/DrawerStyles';
import { navigateToItem } from '../../redux/actions/NavigationAction';

export class _MenuItem extends Component {
  _navigate(route) {
    return this.props.navigation.dispatch(
      NavigationActions.reset({
        index: 0,
        actions: [NavigationActions.navigate({ routeName: `${route}` })],
      }));
  }

  _handleNavigation = () =>Â {
    this._navigate(this.props.navigationRoute, {
      isStatusBarHidden: false,
    });
    this.props.dispatch(navigateToItem(this.props.itemName));
  };

  _getBackgroundColor = () => {
    if (this.props.itemName === this.props.currentTab) {
      return { backgroundColor: '#eaebff' };
    }
    return { backgroundColor: '#FFFFFF' };
  };

  render() {
    return (
      <View style={this._getBackgroundColor()}>
        <TouchableOpacity
          style={styles.menuItem}
          onPress={this._handleNavigation}
        >
          <Icon
            name={this.props.iconName}
            size={this.props.iconSize}
            color="#232c44"
            style={styles.iconStyles}
          />
          <Text style={styles.menuItemText}>{this.props.itemName}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

_MenuItem.propTypes = {
  itemName: PropTypes.string.isRequired,
  iconName: PropTypes.string.isRequired,
  iconSize: PropTypes.number.isRequired,
  navigationRoute: PropTypes.string.isRequired,
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
    dispatch: PropTypes.func.isRequired,
  }).isRequired,
  currentTab: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  return {
    currentTab: state.navigationReducer.currentTab,
  };
};

const MenuItem = connect(mapStateToProps)(_MenuItem);
export default MenuItem;
