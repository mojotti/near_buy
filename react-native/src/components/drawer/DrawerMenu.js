import React, { Component } from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { logout } from '../../redux/actions/AuthorizationAction';
import { styles } from '../../static/styles/DrawerStyles';
import MenuItem from './MenuItem';

class DrawerMenu extends Component {
  render() {
    const allItemsProps = {
      itemName: 'All items',
      iconName: 'public',
      iconSize: 27,
      navigationRoute: 'ItemExplorer',
      navigation: this.props.navigation,
    };
    const myItemProps = {
      itemName: 'My items',
      iconName: 'account-circle',
      iconSize: 27,
      navigationRoute: 'UserItems',
      navigation: this.props.navigation,
    };
    const myChatProps = {
      itemName: 'Chats',
      iconName: 'comment',
      iconSize: 27,
      navigationRoute: 'MyChats',
      navigation: this.props.navigation,
    };

    return (
      <View style={styles.container}>
        <View>
          <MenuItem {...allItemsProps} />
          <MenuItem {...myItemProps} />
          <MenuItem {...myChatProps} />
        </View>

        <TouchableHighlight
          onPress={() => this.props.onLogout()}
          style={styles.logoutButton}
        >
          <Text style={styles.loginText}>Logout</Text>
        </TouchableHighlight>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onLogout: () => {
      dispatch(logout());
    },
  };
};

DrawerMenu.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  onLogout: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(DrawerMenu);
