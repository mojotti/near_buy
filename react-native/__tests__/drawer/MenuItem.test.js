import React from 'react';
import { shallow } from 'enzyme';
import { _MenuItem } from '../../src/components/drawer/MenuItem';

describe('<_MenuItem />', () => {
  test('renders correctly when tab is current tab', () => {
    const menuItem = shallow(
      <_MenuItem itemName={'current'} currentTab={'current'} />
    );

    expect(menuItem).toMatchSnapshot();
  });

  test('renders correctly when tab is not the current tab', () => {
    const menuItem = shallow(
      <_MenuItem itemName={'not current'} currentTab={'current'} />
    );

    expect(menuItem).toMatchSnapshot();
  });

  test('navigates when menu item is pressed', () => {
    const dispatchSpy = jest.fn();
    const menuItem = shallow(
      <_MenuItem
        itemName={'not current'}
        currentTab={'current'}
        dispatch={dispatchSpy}
        navigationRoute="foo"
        navigation={{ dispatch: () => {} }}
      />,
    );

    menuItem.find('TouchableOpacity').simulate('Press');
    expect(dispatchSpy.mock.calls.length).toEqual(1);
  });
});

