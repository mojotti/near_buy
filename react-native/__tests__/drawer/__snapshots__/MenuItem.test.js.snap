// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<_MenuItem /> renders correctly when tab is current tab 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#eaebff",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 2,
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#232c44"
      size={12}
      style={
        Object {
          "paddingRight": 3,
        }
      }
    />
    <Component
      style={
        Object {
          "color": "#232c44",
          "fontFamily": null,
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      current
    </Component>
  </TouchableOpacity>
</Component>
`;

exports[`<_MenuItem /> renders correctly when tab is not the current tab 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#FFFFFF",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 2,
        "padding": 10,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="#232c44"
      size={12}
      style={
        Object {
          "paddingRight": 3,
        }
      }
    />
    <Component
      style={
        Object {
          "color": "#232c44",
          "fontFamily": null,
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      not current
    </Component>
  </TouchableOpacity>
</Component>
`;
